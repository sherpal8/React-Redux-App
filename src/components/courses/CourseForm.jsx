import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput.jsx";
import SelectInput from "../common/SelectInput.jsx";

const CourseForm = ({
  course,
  authors,
  onSave,
  onChange,
  saving = false,
  errors = {}
}) => {
  return (
    <form onSubmit={onSave}>
      <h2>{course.id ? "Edit" : "Add"} Course</h2>
      {errors.onSave && (
        <div className="alert alert-danger" role="alert">
          {errors.onSave}
        </div>
      )}

      <TextInput
        name="title"
        label="Title"
        placeholder="Course title"
        value={course.title}
        onChange={onChange}
        error={errors.title}
      />

      <SelectInput
        name="authorId"
        label="Author"
        value={course.authorId || ""}
        defaultOption="Select Author"
        options={authors.map(author => ({
          value: author.id,
          text: author.name
        }))}
        onChange={onChange}
        error={errors.author}
      />

      <TextInput
        name="category"
        label="Category"
        placeholder="Course category"
        value={course.category}
        onChange={onChange}
        error={errors.category}
      />

      <button type="submit" disabled={saving} className="btn btn-primary">
        {saving ? "Saving" : "Save"}
      </button>
    </form>
  );
};

CourseForm.propTypes = {
  course: PropTypes.object.isRequired,
  authors: PropTypes.array.isRequired,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool,
  errors: PropTypes.object
};

export default CourseForm;
