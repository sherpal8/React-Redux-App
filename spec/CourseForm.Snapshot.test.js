import React from "react";
import CourseForm from "../src/components/courses/CourseForm.jsx";
import renderer from "react-test-renderer";
import { courses, authors } from "../tools/mockData";

it("set submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );
  expect(tree).toMatchSnapshot();
});
