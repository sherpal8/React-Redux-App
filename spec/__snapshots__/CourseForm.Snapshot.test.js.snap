// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set submit button label 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Course
  </h2>
  <div
    className="form-group has-error"
  >
    <label
      htmlFor="title"
    >
      Title
      <div
        className="field"
      >
        <input
          className="form-control"
          label="Title"
          name="title"
          onChange={[MockFunction]}
          placeholder="Course title"
          type="text"
          value="Securing React Apps with Auth0"
        />
      </div>
    </label>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="authorId"
    >
      Author
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="authorId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Author
        </option>
        <option
          value={1}
        >
          Cory House
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group has-error"
  >
    <label
      htmlFor="category"
    >
      Category
      <div
        className="field"
      >
        <input
          className="form-control"
          label="Category"
          name="category"
          onChange={[MockFunction]}
          placeholder="Course category"
          type="text"
          value="JavaScript"
        />
      </div>
    </label>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
